### PR TITLE
show project names as a prompt prefix if we are in a named project

### PR DESCRIPTION
**Pull request by KristofferC**
mar-23-2018 15-59-12

### CODE REVIEW

**Comment by KristofferC** [comment]
The only negative thing is that this causes a little lag when entering the pkg prompt for the first time if running without precompile statements (~2 seconds). Enough to get a bit confused why pressing the button didn't active the mode. But with precompile statements it is good.

**Comment by KristofferC** [comment]
I could just add a `@schedule Types.EnvCache()` in the `__init__` if we are in interactive mode to "steal" downtime in the REPL and use it to compile the `EnvCache` constructor...

**Comment by StefanKarpinski** [comment]
What about the case where the project file has no name and you want to use `basename(dirname(project_file))` as the environment name? E.g. shared environments like `~/.julia/environments/v0.7`.

**Comment by KristofferC** [comment]
>  you want to use `basename(dirname(project_file))` as the environment name

So you want the name of the project to implicitly be the directory-name  it is under? Sure, that can be done easily. Would the "default" prompt then be `(0.7) pkg>`?

**Comment by StefanKarpinski** [comment]
I guess it would be `(v0.7) pkg>` but yes.

**Comment by KristofferC** [comment]
Now uses the folder name if the project is unnamed.

File: docs/src/index.md [hunk 1]
```diff
@@ -185,17 +185,25 @@ global configuration data is saved. Later entries in the depot path are treated
...
 
-The Pkg REPL-mode is entered using from the Julia REPL using the key `]`.
+The Pkg REPL-mode is entered from the Julia REPL using the key `]`.
+
+```
+(v0.7) pkg>
+```
+
+The part inside the parenthesis of the prompt shows the name of the current project.
+Since we haven't created our own projeact yet, we are in the default project, located at `~/.julia/environments/v0.7`
```
[the remark this answers is not in the document]
Comment (KristofferC) [reply docs/src/index.md]: I like project over environment. Why is directory name not accurate?

**Pull request merged**